docs/middleware/BLSPubkeyRegistry.md

```diff
@@ -0,0 +1,36 @@
+
+## Flows
+
+### registerOperator
+
+The RegistryCoordinator for the AVS makes call to the BLSPubkeyRegistry to register an operator with a certain public key for a certain set of quorums. The BLSPubkeyRegistry verifies that the operator in fact owns the public key by [making a call to the BLSPublicKeyCompendium](./BLSPublicKeyCompendium.md#integrations). It then, for each quorum the operator is registering for, adds its public key to the aggregate quorum public key, ends the active block range for the previous aggregate public key, and begins a the active block range for the new aggregate quorum public key. Updates are stored using the following struct:
+```solidity
+/// @notice Data structure used to track the history of the Aggregate Public Key of all operators
+struct ApkUpdate {
+    // first 24 bytes of keccak256(apk_x, apk_y)
+    bytes24 apkHash;
+    // block number at which the update occurred
+    uint32 updateBlockNumber;
+    // block number at which the next update occurred
+    uint32 nextUpdateBlockNumber;
+}
+```
+
+The aggregate quorum public key stored in the contract is also overwritten with the new aggregate quorum public key.
+
+The function also returns the hash of the operator's public key as it may be used as is for the operator in the AVS.
+
+### deregisterOperator
+
+The RegistryCoordinator for the AVS makes call to the BLSPubkeyRegistry to deregister an operator with a certain public key for a certain set of quorums. The BLSPubkeyRegistry verifies that the operator in fact owns the public key by [making a call to the BLSPublicKeyCompendium](./BLSPublicKeyCompendium.md#integrations) (redundant check). It then, for each quorum the operator is registering for, subtracts its public key from the aggregate quorum public key, ends the active block range for the previous aggregate public key, and begins a the active block range for the new aggregate quorum public key. 
+
+The aggregate quorum public key stored in the contract is also overwritten with the new aggregate quorum public key.
+
+Note that the contract does not check that the quorums that the operator's public key is being subtracted from are a subset of the quorums the operator is registered for, that logic is expected to be done in the RegistryCoordinator.
+
+## Integrations 
+
+The main integration with the BLSPubkeyCompendium is used by the AVSs signature checker (TODO: Link docs). An offchain actor provides a public key, a quorum id, and an index in the array of aggregate quorum public key hashes, and the AVS's signature checker verifies that a certain quorum's aggregate public key hash at a certain block number was in fact the hash of the provided public key. Look at `getApkHashForQuorumAtBlockNumberFromIndex`.
```

Comment:
so are "integrations" supposed to be "what this contract integrates with" rather than "how to integrate with this contract"?
If so, I would recommend calling this section something like "Downstream/Upstream Dependencies" instead, depending on if you are talking about what depends on this contract / what it depends on.

Seems like there's also a typo where you have `BLSPubkeyCompendium` instead of `BLSPubkeyRegistry` ?